### PR TITLE
Remove Legacy Dune Client

### PR DESCRIPTION
**Pull request by bh2smith**
As a second step to using the new Dune Client, we also phase out the Legacy Dune Client entirely.

# Test Plan

No logical changes, just existing CI - now that the duneapi requirement has been removed.

### CODE REVIEW

File: tests/unit/test_implementations.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class TestQueryMonitor(unittest.TestCase):
     def setUp(self) -> None:
         self.date = datetime.datetime(year=1985, month=3, day=10)
         self.query_params = [
-            QueryParameter.enum_type("Enum", "option1", ["option1", "option2"]),
```
Comment (gentrexha) [inline tests/unit/test_implementations.py:19]: Why was the list with options 1 and 2 removed?

File: tests/unit/test_models.py [hunk 2]
```diff
@@ -101,7 +101,6 @@ def test_as_query_params(self):
                 QueryParameter.enum_type(
                     name="TimeUnits",
                     value=str(TimeUnit.MINUTES.value),
-                    options=TimeUnit.options(),
```
Comment (gentrexha) [inline tests/unit/test_models.py:104]: Also same question here: Why was this removed?

**Review by gentrexha — APPROVED** [review]
LGTM!

Just 2 questions I'm curious about.

File: tests/unit/test_implementations.py (hunk 1 above)
Comment (bh2smith) [reply tests/unit/test_implementations.py:19]: So. In the legacy version we supported upsert query (I.e altering the query). The new dune client won't support this for a while and when we aren't changing the query, we don't need to provide the enum options... just the value. This is also the reason why we would have seen a type clash with the QueryParameters class.

**Comment by bh2smith** [comment]
@gentrexha - does Approving this mean you also approve the PR its based on #30 ?

**Comment by gentrexha** [comment]
> @gentrexha - does Approving this mean you also approve the PR its based on #30 ?

I'm going through that now.
